libp2p/utils.py

```diff
@@ -95,3 +98,25 @@ def encode_fixedint_prefixed(msg_bytes: bytes) -> bytes:
     len_bytes = await reader.read(SIZE_LEN_BYTES)
     len_int = int.from_bytes(len_bytes, "big")
     return await reader.read(len_int)
+
+
+TItem = TypeVar("TItem")
+
+
+class IQueue(Generic[TItem]):
+    async def put(self, item: TItem):
+        ...
+
+    async def get(self) -> TItem:
+        ...
+
+
+class TrioQueue(IQueue):
```

Comment:
I'm inclined to :-1: this approach.

https://trio.readthedocs.io/en/latest/reference-io.html#trio.abc.Channel

First, the `trio.abc.Channel` API is this so I'd advocate using that API instead or at least that name.  You *may* have to actually create that class because I haven't looked to see if `trio` provides an implementation.

Reply:
Big thanks for the review! `TrioQueue` is used temporary just to make the code running. I found `trio.abc.Channel` is extremely helpful for us to replace the [`event.wait` patterns in our original implementation](https://github.com/libp2p/py-libp2p/commit/b6627d1ca32d13ad27fc96d723b5f0ab46eb6228#diff-d01e8ca42064b756a3da55aa43f9aff3L61-L103).